### PR TITLE
Add Article service using Summon gem

### LINKED ISSUE #40: Add Articles (Summon) Service to the API
## Description
Summon is a index of Article content the Library subscribes that Ex Libris maintains and provides via an API. It is labeled "Articles+} on the website. 

### Source Service End User URL
https://princeton.summon.serialssolutions.com/#!/

### Supporting Service Documentation
https://developers.exlibrisgroup.com/summon/apis/
https://github.com/summon/summon.rb (Gem for the API)

### Connectivity
Need Client ID and Secret Key. In last pass as the Summon Over Alma key. See https://github.com/pulibrary/discoveryutils/blob/main/classes/Summon/Summon.php#L275 for the current authentication implementation. 

### Acceptance Criteria

- [x] Returns the results to the equivalent of  a basic keyword query of the service
- [x] Provides Search results with the same default sort as the native service
  interface 
- [x] API response should include a count of the total results
- [x] API response should include a link to the user interface for the source service, so that a user can continue their search there
- [x] API response should include the first 3 results from the source
  application
- [x] API response should include the following fields
    * Title Field
    * Individual URL for each result returned
    * Format
    * Abstract
    * Publication Title
    * Publication Date
    * Publication Year
    * Snippet (highlighted query results)
    * ISxN
    * Full Text Available
  

### Implementation Notes
Current JSON Response: https://library.princeton.edu/utils/articles/any?query=cats&sid=library-main-summon
Current implemetation: https://github.com/pulibrary/discoveryutils/tree/main/classes/Summon
Current "holdings" value is irrelevant

### PR DESCRIPTION
**Pull request by maxkadel**
Closes #40 

See also [Princeton Ansible PR](https://github.com/pulibrary/princeton_ansible/pull/4379) to add required environment variable

- [X] Limit to three responses
- [X] Sanitize inputs
- [X] Update query to match query on existing site
- [X] Add journal-specific metadata to other fields
- [x] Add Swagger documentation

### CODE REVIEW

**Comment by maxkadel** [comment]
Deployed to staging 10/9/2023 10:58 am EDT

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,6 @@
 
 .byebug_history
 /solr/*
+
+# Secrets for development
+.env*
```
Comment (sandbergja) [inline .gitignore:30]: Nice!

File: app/models/article_document.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  delegate :abstract, to: :document
+  delegate :end_page, to: :document
+  delegate :issue, to: :document
+  delegate :publication_title, to: :document
+  delegate :publisher, to: :document
+  delegate :start_page, to: :document
+  delegate :volume, to: :document
+
+  def title
+    full_sanitizer = Rails::HTML5::FullSanitizer.new
```
Comment (sandbergja) [inline app/models/article_document.rb:15]: This is nice!  Is the sanitization covered by any tests?

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel !  One testing question, but it is not a blocker

File: app/models/article_document.rb (hunk 2 above)
Comment (maxkadel) [reply app/models/article_document.rb:15]: You're right! Added!

**Pull request merged**